nncf/common/tensor_statistics/statistics_serializer.py

```diff
@@ -33,87 +35,128 @@ def load_metadata(dir_path: Path) -> Dict[str, Any]:
     mapping = metadata.get("mapping", {})
 
+    load_file_func = get_load_file_method(backend)
     for statistics_file in path.iterdir():
         if statistics_file.name == METADATA_FILE:
             continue  # Skip the metadata file
 
         try:
-            fail_if_symlink(statistics_file)
-            with gzip.open(statistics_file, "rb") as f:
-                sanitized_name = statistics_file.name
-                original_name = mapping.get(sanitized_name, sanitized_name)
-                statistics[original_name] = pickle.load(f)
-        except (pickle.UnpicklingError, IOError) as e:
+            sanitized_name = statistics_file.name
+            original_name = mapping.get(sanitized_name, sanitized_name)
+            statistics[original_name] = load_file_func(statistics_file)
+        except Exception as e:
             raise nncf.InternalError(f"Error loading statistics from {statistics_file.name}: {e}")
-    return statistics, metadata.get("metadata", {})
+
+    return statistics, {key: value for key, value in metadata.items() if key != "mapping"}
 
 
 def dump_to_dir(
-    statistics: Dict[str, Any], dir_path: str, additional_metadata: Optional[Dict[str, Any]] = None
+    statistics: Dict[str, Dict[str, TTensor]],
+    dir_path: str,
+    backend: TensorBackend,
+    additional_metadata: Optional[Dict[str, Any]] = None,
 ) -> None:
     """
-    Dumps statistics to gzip-compressed files in the specified directory, while maintaining a mapping file.
+    Saves statistics and metadata to a directory.
     :param statistics: A dictionary with statistic names as keys and the statistic data as values.
     :param dir_path: The path to the directory where the statistics will be dumped.
+    :param backend: Backend type to determine the saving function.
     :param additional_metadata: A dictionary containing any additional metadata to be saved with the mapping.
     """
     path = Path(dir_path)
     path.mkdir(parents=True, exist_ok=True)
 
-    metadata, mapping = {}, {}
+    metadata: Dict[str, Any] = {"mapping": {}}
 
+    save_file_func = get_save_file_method(backend)
     for original_name, statistics_value in statistics.items():
         sanitized_name = sanitize_filename(original_name)
         file_path = path / sanitized_name
 
         # Update the mapping
-        mapping[sanitized_name] = original_name
+        metadata["mapping"][sanitized_name] = original_name
 
         try:
-            fail_if_symlink(file_path)
-            with gzip.open(file_path, "wb") as f:
-                pickle.dump(statistics_value, f)
-        except (IOError, pickle.PicklingError) as e:
+            save_file_func(statistics_value, file_path)
+        except Exception as e:
             raise nncf.InternalError(f"Failed to write data to file {file_path}: {e}")
 
-    # Add additional metadata if provided
     if additional_metadata:
-        metadata["metadata"] = additional_metadata
+        metadata |= additional_metadata
```

Comment:
Where i can find description of metadata structure?

Reply:
Added a small description at the file begginig